app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java

```diff
@@ -186,30 +179,42 @@ public void rememberEulaAccepted(Context context) {
         editor.commit();
     }
 
-    /**
-     * User SDK function to login
-     */
-    public void loginToDhis(String serverUrl, String username, String password) {
-        //Delegate real login attempt to parent in sdk
-        //// FIXME: 28/12/16
-        //super.login(serverUrl, username, password);
+    public void login(String serverUrl, String username, String password) {
+        Credentials credentials = new Credentials(serverUrl, username, password);
+
+        mLoginUseCase.execute(credentials, new ALoginUseCase.Callback() {
+            @Override
+            public void onLoginSuccess() {
+                mLoginActivityStrategy.finishAndGo();
+            }
+
+            @Override
+            public void onServerURLNotValid() {
+                serverText.setError("Server url not valid");
+                showError("Server url not valid");
+            }
+
+            @Override
+            public void onInvalidCredentials() {
+                showError("Invalid credentials");
+            }
+
+            @Override
+            public void onNetworkError() {
+                showError("There is a problem with the network, try again later.");
+            }
+        });
     }
 
-    /**
-     * This logout is called from the success user autentication, and try to login in the server
-     * with the correct userdata.
-     */
-    //// FIXME: 28/12/16
-    //@Subscribe
-    public void onLogoutFinished() {
-        SdkLoginController.logInUser(serverUrl, ServerAPIController.getSDKCredentials());
+    public void showError(String message) {
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
     }
 
+
     //// FIXME: 28/12/16
     //@Subscribe
     public void onLoginFinished() {
-        /*
-        if (result != null && result.getResourceType().equals(ResourceType.USERS)) {
+/*        if (result != null && result.getResourceType().equals(ResourceType.USERS)) {
```

Comment:
I don´t remove for remember FIXME in future, this method do something interesting to remenber